OpenRA.Game/TextNotificationsManager.cs

```diff
@@ -28,24 +28,46 @@ static TextNotificationsManager()
 				systemMessageLabel = "Battlefield Control";
 		}
 
+		public static void AddFeedbackLine(string text)
+		{
+			AddTextNotification(TextNotificationPool.Feedback, systemMessageLabel, text, systemMessageColor, systemMessageColor);
+		}
+
 		public static void AddSystemLine(string text)
 		{
 			AddSystemLine(systemMessageLabel, text);
 		}
 
-		public static void AddSystemLine(string name, string text)
+		public static void AddSystemLine(string prefix, string text)
 		{
-			Game.OrderManager.AddChatLine(name, systemMessageColor, text, systemMessageColor);
+			AddTextNotification(TextNotificationPool.System, prefix, text, systemMessageColor, systemMessageColor);
 		}
 
-		public static void AddChatLine(string name, Color nameColor, string text)
+		public static void AddChatLine(string prefix, string text, Color? prefixColor = null, Color? textColor = null)
 		{
-			Game.OrderManager.AddChatLine(name, nameColor, text, chatMessageColor);
+			AddTextNotification(TextNotificationPool.Chat, prefix, text, prefixColor, textColor);
 		}
 
 		public static void Debug(string s, params object[] args)
 		{
 			AddSystemLine("Debug", string.Format(s, args));
 		}
+
+		static void AddTextNotification(TextNotificationPool pool, string prefix, string text, Color? prefixColor = null, Color? textColor = null)
+		{
+			var thePrefixColor = prefixColor ?? chatMessageColor;
```

Comment:
Style nit: we don't use "the" prefixes anywhere else in the code so these stick out as looking a bit odd.
These two lines are quite short and only used once, so how about we just inline them directly below? That would also save us from evaluating them if the pool is not enabled.